File: server/Domain/Enums/Widget/WidgetType.cs
using System.Text.Json.Serialization;

namespace Domain;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum WidgetType
{
    TaskList,
    FolderList,
    ActivityFeed,
    NotificationSummary,
    UpcomingDeadlines,

    // Insights / analytics
    WorkspaceHealth,
    WorkloadSummary,
    GoalProgress,

    // Utility
    QuickActions,
    Calendar,

    // Visual / optional
    Hero,

}

File: frontend/src/types/widget-type.ts
export const WidgetType = {
  TaskList: "TaskList",
  FolderList: "FolderList",
  ActivityFeed: "ActivityFeed",
  NotificationSummary: "NotificationSummary",
  UpcomingDeadlines: "UpcomingDeadlines",
  WorkspaceHealth: "WorkspaceHealth",
  WorkloadSummary: "WorkloadSummary",
  GoalProgress: "GoalProgress",
  QuickActions: "QuickActions",
  Calendar: "Calendar",
  Hero: "Hero",
} as const;

export type WidgetType = (typeof WidgetType)[keyof typeof WidgetType];
